sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -498,7 +498,10 @@ class Dataset[T] private[sql](
    * @since 1.6.0
    */
   def explain(extended: Boolean): Unit = {
-    val explain = ExplainCommand(queryExecution.logical, extended = extended)
+    // Because views are possibly resolved in the analyzed plan of this dataset. We use analyzed
+    // plan in `ExplainCommand`, for consistency. Otherwise, the plans shown by explain command
+    // might be inconsistent with the evaluated data of this dataset.
+    val explain = ExplainCommand(queryExecution.analyzed, extended = extended)
```

Comment:
Thank you for pinging me, @viirya . I like this PR. Could you add a test case for this please?

Reply:
Sure. I originally thought manual test might be enough. Will add one unit test soon.

Reply:
Thank you. Yes. This is important fix. I want to prevent the future regressions.

Reply:
Added an unit test. Thanks for review.